Urgent fix coil type

athoynilimanew opened the pull request:
## Linked Issues

Fixes duplication of CoilType Enum

Closes #{ID}

## Description

<!-- What is your PR trying to achieve? How did you go about achieving it? -->

## Interface Changes

<!-- If you've had to update an interface or introduce a new interface as part of your change then let us know here. -->

## Checklist

I confirm that I have completed the following checks:

- [ ] Tests run locally and pass `pytest tests --reactor`
- [ ] Code quality checks run locally and pass `pre-commit run --from-ref develop --to-ref HEAD`
- [ ] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

geograham commented on bluemira/equilibria/coils/coil_type.py:
Could you write a simple test for the CoilType class?

athoynilimanew replied:
added in the latest commit.

geograham commented on bluemira/equilibria/coils/_coil.py:
This is probably a silly question, but could you explain why CoilType needed to be moved to a separate file? Is it a circular import error related to plotting.py?

athoynilimanew replied:
Exactly. Somehow plotting.py and pf_coil.py were talking to each other and it was the source of circular error

je-cook reviewed (changes requested):
This is not ready. I think there were a few things missing from the review of #3021 on top of that. I will start with comments there and come back here.

je-cook reviewed (changes requested):
On top of the comments in #3021 I think the coils enum should be moved to base. I think moving it either to constants.py or its own file in base. I'm unsure of the exact location but generally inside the base module

athoynilimanew commented:
> On top of the comments in #3021 I think the coils enum should be moved to base. I think moving it either to constants.py or its own file in base. I'm unsure of the exact location but generally inside the base module

Moved to constants.py and accordingly changed the imports. moved test function as well

athoynilimanew commented:
> This is not ready. I think there were a few things missing from the review of #3021 on top of that. I will start with comments there and come back here.

Applied/responded to all required changes (see #3021 for additional comments)

je-cook commented on bluemira/equilibria/equilibrium.py:
```diff
@@ -1073,13 +1073,8 @@ def set_vcontrol(self, vcontrol_str: Optional[str] = None):
             if vcontrol_type is VerticalPositionControlType.VIRTUAL:
                 self.controller = VirtualController(self, gz=2.2)
-            elif vcontrol_type is VerticalPositionControlType.FEEDBACK:
+            if vcontrol_type is VerticalPositionControlType.FEEDBACK:
```
I think probably more robust is the below as we haven't dealt with any other control type
```suggestion
            else:
```

je-cook commented on bluemira/geometry/coordinates.py:
```diff
@@ -628,32 +630,29 @@ def rotation_matrix(theta: float, axis: Union[str, np.ndarray] = "z") -> np.ndar
-            raise CoordinatesError(
-                f"Incorrect rotation axis: {axis}\nplease select from: ['x', 'y', 'z']"
-            )
+
```
```suggestion
         else:
             raise NotImplementedError
```

je-cook commented on bluemira/geometry/tools.py:
```diff
@@ -727,18 +729,13 @@ def convex_hull_wires_2d(
+    elif hull_plane is HullPlane.XY:
         plane_idxs = (0, 1)
-    elif hull_plane is HullPlaneType.YZ:
+    else:  # YZ
         plane_idxs = (1, 2)
```
```suggestion
    elif hull_plane is HullPlaneType.YZ:
        plane_idxs = (1, 2)
    else: 
        raise NotImplementedError
```
